client/my-sites/email/email-management/home/email-plan-warnings.jsx

```diff
@@ -61,19 +61,32 @@ class EmailPlanWarnings extends Component {
 	}
 
 	render() {
-		const { warning } = this.props;
-		if ( ! warning ) {
+		const { domain, translate, warning } = this.props;
+
+		if ( ! warning && domain.currentUserCanAddEmail ) {
 			return null;
 		}
 
 		return (
 			<div className="email-plan-warnings__container">
-				<div className="email-plan-warnings__warning">
-					<div className="email-plan-warnings__message">
-						<span>{ warning.message }</span>
+				{ ! domain.currentUserCanAddEmail && (
+					<div className="email-plan-warnings__warning">
+						<div className="email-plan-warnings__message">
+							<span>
+								{ translate( 'Only the email subscription owner can add or remove mailboxes.' ) }
```

Comment:
We show the following warning on the `Purchase Settings` page when a subscription is owned by another user:

![image](https://user-images.githubusercontent.com/594356/172177676-aaade107-4d49-4d11-8431-b49590f54ec7.png)

I'm not sure if it's worth it but we could refine this new warning so it is more consistent with it:

> This mailbox was/Those mailboxes were purchased by a different WordPress.com account. To add or remove mailboxes, log in to that account or contact the account owner.

![image](https://user-images.githubusercontent.com/594356/172179320-987bccf0-14e4-41ce-8d9a-6d1d7b25d933.png)

Reply:
Good stuff! 

What is the exact logic behind the `currentUserCanAddEmail` attribute? In the WordPress.com support docs, we [say this](https://wordpress.com/support/add-email/adding-professional-email-to-your-site/#purchase-professional-email-for-an-existing-domain):

> Email can only be added by the user that pays for the domain subscription.

Is that what it comes down to – only the domain owner can purchase email and then only the email owner can add/remove mailboxes? That means that the domain owner and the email owner *has to be* the same user.

Reply:
I think that's right. For your information, the logic is located in the `Domain_Email_Eligibility::can_user_add_email_for_domain_subscription()` method in the backend. It checks several things regarding the domain in addition to the owner of the domain subscription.